File: FrontEnd/types/ticket.ts
// Ticket / Helpdesk domain types

export type TicketStatus = 'Open' | 'In Progress' | 'Resolved' | 'Closed' | 'Pending';
export type TicketPriority = 'Critical' | 'High' | 'Medium' | 'Low';

export interface Ticket {
  id: string;
  subject: string;
  hotel: string;
  status: TicketStatus | string;
  priority: TicketPriority | string;
  assignedTo: string;
  createdAt: string;
  lastUpdate: string;
  category?: string;
  description?: string;
}

export interface TicketMessage {
  id: string;
  sender: string;
  content: string;
  timestamp: string;
  isAgent?: boolean;
}

export interface TicketNote {
  id: string;
  author: string;
  content: string;
  timestamp: string;
}

File: BackEnd/app/models/ticket.py
from sqlalchemy import Column, Integer, String, Boolean, ForeignKey, DateTime
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.database import Base


class Ticket(Base):
    __tablename__ = "tickets"

    id = Column(Integer, primary_key=True, index=True)
    subject = Column(String, index=True)
    description = Column(String)
    priority = Column(String)  # Low, Medium, High, Critical
    status = Column(
        String, default="Open"
    )  # Open, In Progress, Waiting on Client, Resolved, Closed
    category = Column(String)  # Technical, Billing, Feature Request, Other

    tenant_id = Column(UUID(as_uuid=True), ForeignKey("tenants.id"))
    tenant = relationship("Tenant")

    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
